packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -698,3 +708,59 @@ export function isAzureCoreModel(t: SdkType): boolean {
 export function isPagedResultModel(context: TCGCContext, t: SdkType): boolean {
   return context.__pagedResultSet.has(t);
 }
+
+/**
+ * Find corresponding http parameter list for a client initialization or service method parameter
+ * @param method
+ * @param param
+ * @returns
+ */
+export function getHttpOperationParameter(
+  method: SdkServiceMethod<SdkHttpOperation>,
+  param: SdkMethodParameter,
+): (
+  | SdkPathParameter
+  | SdkQueryParameter
+  | SdkHeaderParameter
+  | SdkCookieParameter
+  | SdkBodyParameter
+)[] {
+  const operation = method.operation;
+  const result: (
+    | SdkPathParameter
+    | SdkQueryParameter
+    | SdkHeaderParameter
+    | SdkCookieParameter
+    | SdkBodyParameter
+  )[] = [];
+  const queue: SdkModelPropertyType[] = [param];
+  const visited: Set<SdkModelType> = new Set();
+  while (result.length === 0 && queue.length > 0) {
+    const param = queue.pop();
+    for (const p of operation.parameters) {
+      for (const cp of p.correspondingMethodParams) {
+        if (cp === param) {
+          result.push(p);
+        }
+      }
+    }
+    if (operation.bodyParam) {
+      for (const cp of operation.bodyParam.correspondingMethodParams) {
+        if (cp === param) {
+          result.push(operation.bodyParam);
+        }
+      }
+    }
+    if (result.length === 0 && param?.type.kind === "model" && !visited.has(param.type)) {
```

Comment:
Is this a case of spread?

Reply:
Seems grouping? Again, better to add comment.

Reply:
it is for `@bodyRoot`, for example:
```
import "@typespec/http";
using Http;
model TestRequest {
  @header
  h: string;

  @query
  q: string;

  prop1: string;
  prop2: string;
}
op test(@bodyRoot request: TestRequest): void;
```

Reply:
added a test with `@override`.